src/components/core/Markdown/Link.tsx

```diff
@@ -0,0 +1,73 @@
+import IconContainer from './IconContainer';
+import { useCallback } from 'react';
+
+export default function Bold({
+  selectedRange,
+  textAreaRef,
+  setSelectedRange,
+  setUserDragged,
+}: {
+  selectedRange: number[];
+  textAreaRef: React.RefObject<HTMLTextAreaElement>;
+  setSelectedRange: (value: number[] | ((prevVar: number[]) => number[])) => void;
+  setUserDragged: (value: boolean) => void;
+}) {
+  const handleClick = useCallback(() => {
+    setUserDragged(false);
+
+    const textArea = textAreaRef.current;
+    if (textArea) {
+      const [start, end] = selectedRange;
+      const selectedText = textArea.value.substring(start, end);
+
+      // If user is inbetween two brackets, they probably just clicked the link button. Do nothing
+      if (textArea.value.substring(start - 1, end + 1) == '[]') {
+        return;
+      }
+      if (selectedText.length > 0) {
+        // If the selected text is not already a link, add it
+        const newText =
+          textArea.value.substring(0, start) +
+          `[${selectedText}](https://)` +
+          textArea.value.substring(end);
+        textArea.value = newText;
+        setSelectedRange([start + 11 + selectedText.length, start + 11 + selectedText.length]); // Link [0, 4] -> [Link](https://) (end)
```

Comment:
love that the cursor ends up right at the beginning of the url, ready for you to type the rest - super useful!